packages/plugins/src/plugins/aave/helpers.ts

```diff
@@ -338,3 +355,99 @@ export function getElementLendingData(
   };
   return elementData;
 }
+
+export function getRates(
+  tokenAddress: string,
+  platformId: string,
+  formattedReserve: Pick<
+    FormattedReserve,
+    'supplyAPR' | 'variableBorrowAPR' | 'totalLiquidity' | 'totalDebt'
+  >,
+  elementName: string
+): BorrowLendRate {
+  const lendingApr = Number(formattedReserve.supplyAPR);
+  const borrowingApr = Number(formattedReserve.variableBorrowAPR);
+  const depositedAmount = Number(formattedReserve.totalLiquidity);
+  const borrowedAmount = Number(formattedReserve.totalDebt);
+  return {
+    tokenAddress,
+    borrowYield: {
+      apy: aprToApy(borrowingApr),
+      apr: borrowingApr,
+    },
+    borrowedAmount,
+    depositYield: {
+      apy: aprToApy(lendingApr),
+      apr: lendingApr,
+    },
+    depositedAmount,
+    platformId,
+    poolName: elementName,
+  };
+}
+
+export function getTokenPriceSourceFromReserve(
+  aTokenAddress: string,
+  networkId: NetworkIdType,
+  platformId: string,
+  formattedReserve: Pick<FormattedReserve, 'decimals' | 'underlyingAsset'>,
+  underlyingAssetPrice: TokenPrice,
+  elementName: string
+) {
+  const source: TokenPriceSource = {
+    id: platformId,
+    weight: 1,
+    address: aTokenAddress,
+    networkId,
+    platformId,
+    decimals: formattedReserve.decimals,
+    price: underlyingAssetPrice.price,
+    underlyings: [
+      {
+        address: formattedReserve.underlyingAsset,
+        amountPerLp: 1,
+        decimals: formattedReserve.decimals,
+        networkId,
+        price: underlyingAssetPrice.price,
+      },
+    ],
+    elementName,
+    timestamp: Date.now(),
+  };
+  return source;
+}
+
+// typeguard
+export function isUserReserveData(
+  item: UserReserveData | UserReserveDataV3
+): item is UserReserveData {
+  return 'stableBorrows' in item && 'stableBorrowAPR' in item;
+}
+
+export async function getBalancesAndTotalRewards(
+  owner: Address,
+  addresses: Address[],
+  networkId: EvmNetworkIdType
+): Promise<(bigint | null)[]> {
+  const client = getEvmClient(networkId);
+  const balances = await client.multicall({
+    contracts: addresses.flatMap<ContractFunctionConfig<typeof stkAbi>>((a) => [
+      {
+        abi: stkAbi,
+        address: a,
+        functionName: 'balanceOf',
+        args: [owner],
+      },
+      {
+        abi: stkAbi,
+        address: a,
+        functionName: 'getTotalRewardsBalance',
+        args: [owner],
+      },
+    ]),
+  });
+
+  return balances.map((b) =>
+    !b.result || b.result === zeroBigInt ? null : b.result
```

Comment:
```suggestion
    !b.result || b.result === zeroBigInt ? undefined : b.result
```

Could `undefined` be used here to move away from using `null` ?